### ISSUE TITLE
No live stream if i start systemctl homebridge

### ISSUE THREAD

**Issue by warcanoid** [KhaosT/homebridge-camera-ffmpeg#202]
If I start homebridge systemctl-autoboot I do not get live stream-just loading, everything else works fine, also snapshot every 10s.
But if I stop systemctl homebridge, and start manually, just type homebridge, the live stream works! 
Why autoboot systemctl will not play live stream?

Regards

**Comment by llemtt** [comment]
Different user permissions?

Try manually with sudo -u "autoboot user", does it work?

**Comment by warcanoid** [comment]
No, it is:  sudo: unknown user: autoboot user
Any other suggestion?

**Comment by llemtt** [comment]
Ok, what's the user you have configured to run homebridge with systemctl?

Have you created an "homebridge" user/group?

**Comment by warcanoid** [comment]
First, thank you very much for your help!
I think user is default pi!? I followed this guide: https://timleland.com/setup-homebridge-to-start-on-bootup/#comment-11084
There is under point 3. Create a user to run service: sudo useradd --system homebridge

**Comment by llemtt** [comment]
Try to manually launch homebridge like this to simulate what systemctl does:

sudo -u homebridge homebridge

it should fail launching ffmpeg!

That guide is ok to run homebridge but to let it launch ffmpeg I think you have to add at least the following:

sudo usermod -a -G video homebridge

**Comment by warcanoid** [comment]
JUHUUHUHUH it works!!! Thank you very much! You are the man! I have struggled 2 days! THANKS!

**Comment by warcanoid** [comment]
Hey, can you help again with notifications to get from motion sensor on camera or from d-link motion sensor?

**Comment by jaykyll** [comment]
Hello,

Had the same problem and apply @llemtt command but always same problem :-/
Can you help me?
Thanks

**Comment by llemtt** [comment]
Check that user homebridge has read/write permissions on directories where is files are

**Comment by jaykyll** [comment]
I followed instructions to install basically homebridge, plugins nest, nest-cam, ffmpeg and ffmpeg-omx. In addition I also like @warcanoid followed same tuto to start automatically homebridge with systemctl. So that's why I'v applied the command you list but with no changes. Don't know where to search :-S.

**Comment by warcanoid** [comment]
@jaykyll follow this guide

**Comment by llemtt** [comment]
> I followed same tuto to start automatically homebridge with systemctl. 

so files are supposed to be in /var/homebridge (I have mines in /var/lib/homebridge)

check that user homebridge can read/write that directory

also put "-v verbose" in front of your source configuration in order to get more debug info from ffmpeg

**Comment by jaykyll** [comment]
Thanks @warcanoid, that's exactly what I did. All is working fine but not live stream.

**Comment by llemtt** [comment]
better

ls -laR /var/homebridge/

**Comment by jaykyll** [comment]
Thanks : 

ls -laR /var/homebridge/
/var/homebridge/:
total 20
drwxrwxrwx  4 root       root       4096 jan 16 01:44 .
drwxr-xr-x 12 root       root       4096 jan  5 12:33 ..
drwxr-xr-x  2 homebridge homebridge 4096 jan 14 15:34 accessories
-rwxrwxrwx  1 root       root        654 jan 16 01:09 config.json
drwxrwxrwx  2 root       root       4096 jan 14 23:59 persist

/var/homebridge/accessories:
total 12
drwxr-xr-x 2 homebridge homebridge 4096 jan 14 15:34 .
drwxrwxrwx 4 root       root       4096 jan 16 01:44 ..
-rw-r--r-- 1 homebridge homebridge    2 jan 16 16:41 cachedAccessories

/var/homebridge/persist:
total 48
drwxrwxrwx 2 root       root       4096 jan 14 23:59 .
drwxrwxrwx 4 root       root       4096 jan 16 01:44 ..
-rw-r--r-- 1 homebridge homebridge  514 jan 16 16:41 AccessoryInfo.66224F415F13.json
-rwxrwxrwx 1 root       root        625 jan  5 12:33 AccessoryInfo.CC223DE3CE30.json
-rwxrwxrwx 1 root       root        624 jan 16 16:41 AccessoryInfo.CC223DE3CE40.json
-rw-r--r-- 1 homebridge homebridge 2522 jan 14 23:59 IdentifierCache.66224F415F13.json
-rwxrwxrwx 1 root       root       9646 jan  5 12:33 IdentifierCache.CC223DE3CE30.json
-rwxrwxrwx 1 root       root       9646 jan  5 12:33 IdentifierCache.CC223DE3CE40.json

**Comment by llemtt** [comment]
your files permissions are ok, so set the "debug" to true in your configuration, put "-v verbose" at the start of your "source" and take a look at the log to see what's happening

**Comment by jaykyll** [comment]
Ok I set (and uncomment) the debug=true in homebridge file. But what do you mean by "start of your source" ?

**Comment by llemtt** [comment]
at the beginning of your "source" config like this

"source": "-v verbose ...

**Comment by jaykyll** [comment]
the config.json so? (sorry for misunderstanting :/)

**Comment by llemtt** [comment]
yes same place as "debug"

**Comment by jaykyll** [comment]
ok like this so : 
# Defaults / Configuration options for homebridge
# The following settings tells homebridge where to find the config.json file and where to persist the data (i.e. pair$
HOMEBRIDGE_OPTS=-U /var/homebridge

# If you uncomment the following line, homebridge will log more
# You can display this via systemd's journalctl: journalctl -f -u homebridge
DEBUG=true -v verbose

**Comment by llemtt** [comment]
sorry don't understand what you are editing...

in your installation configuration file should be

/var/homebridge/config.json

**Comment by jaykyll** [comment]
ok but the debug line is in the file /etc/default/homebridge (https://timleland.com/setup-homebridge-to-start-on-bootup/#comment-11084)

**Comment by jaykyll** [comment]
ah ok so : 

in my config.json : 
"platform": "Nest-cam",
                "access_token" : ""
                "debug" : true,
                "useOMX": true

**Comment by llemtt** [comment]
but are you using camera-ffmpeg?

**Comment by jaykyll** [comment]
yes but don't need to set a platform ffmpeg.
when launching homebridge (without systemctl) it works and it's the same config.json.

**Comment by thesynthetic** [comment]
Just had this same issue and it turned out to be caused but the two points below: 

1. The `config.json` file was incorrectly configured and I had to update the `source` and `stillImageSource` to a working `ffmpeg` command that sources video from `/dev/video0` (I think this may also have had to do with the fact that there are two config.json files, one in my home directory and the other in the Homebridge users's home directory)
2. The `homebridge` user did not have permission to access `/dev/video0` which is my USB webcam. This was solved by: `sudo usermod -a -G video homebridge`
